### PR TITLE
chore: code cleaning

### PR DESCRIPTION
**Pull request by paulushcgcj**
---

Thanks for the PR!

Deployments, as required, will be available below:
Any successful deployments (not always required) will be available [here](https://nr-forest-client-22-frontend.apps.silver.devops.gov.bc.ca/)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-forest-client/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-forest-client/actions/workflows/merge.yml)

### CODE REVIEW

**Review by mamartinezmejia — APPROVED** [review]

File: backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java [hunk 1]
```diff
@@ -17,15 +17,28 @@
...
+  // ObjectMapper provides functionality for reading and writing JSON
   private final ObjectMapper mapper;
 
+  /**
+   * This method is called before the entity is saved in the database.
+   * It converts the matchers field of the entity to a JSON string.
+   *
+   * @param entity The entity that is about to be saved in the database.
+   * @param table The table where the entity will be saved.
+   * @return The entity with the converted matchers field.
```
Comment (fterra-encora) [inline backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java:40]: Should we say:
@return **A Publisher that emits** the entity with the converted matchers field.
instead?
I got this by looking at another class in this same PR.

File: backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java [hunk 2]
```diff
@@ -34,13 +47,27 @@ public Publisher<SubmissionMatchDetailEntity> onBeforeConvert(
     return Mono.justOrEmpty(entity.withMatchingField(convertTo(entity)));
   }
 
+  /**
+   * This method is called after the entity is retrieved from the database.
+   * It converts the matchingField field of the entity from a JSON string to a Map.
+   *
+   * @param entity The entity that was retrieved from the database.
+   * @param table The table where the entity was retrieved from.
+   * @return The entity with the converted matchingField field.
```
Comment (fterra-encora) [inline backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java:56]: Same question, should we start @return with "A Publisher that emits ..."?

**Review by fterra-encora — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java (hunk 1 above)
Comment (paulushcgcj) [reply backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java]: That makes complete sense. Updated the docs

File: backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java (hunk 2 above)
Comment (paulushcgcj) [reply backend/src/main/java/ca/bc/gov/app/converters/SubmissionMatchDetailEntityBeforeConvert.java]: That makes complete sense. Updated the docs

**Review by fterra-encora — APPROVED** [review]

**Pull request merged**